Mentions ordering

ronaldlangeveld opened the pull request:
refs https://github.com/TryGhost/Team/issues/2498

- orders mentions by timestamp.

ronaldlangeveld committed:
Added API sorting

ronaldlangeveld committed:
Added Mentions ordering

allouis reviewed (commented):
I don't think the `waitForOneSecond` approach is a good pattern as it's gonna slow down tests.

Instead we can either:
- Use `sinon.useFakeTimers`, this allows us to fake the time and tick it forward
- Don't use `processWebmention` instead create `Mention`s directory and save them in the repository.

ronaldlangeveld committed:
Improved tests

refs https://github.com/TryGhost/Team/issues/2498

- Replaced `useTimeout()` with `sinon.useFakeTimers()` to improve speed
  and avoid flaky tests.
- removed comment
